### PR TITLE
Create sbt submodules per scala version

### PR DESCRIPTION
**Pull request by kyri-petrou**
This allows us to run the compile / test all modules for a given Scala version which makes dev experience a bit better. e.g., to run all the test suites for Scala 3, all we need to run is `sbt ++3.3.1 rootJVM3/test`.

In addition to improving dev experience, it also makes CI run faster as SBT can parallelise the compiling of SBT modules & test execution

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -104,6 +100,51 @@ lazy val root = project
...
+  .aggregate(allProjects *)
+
+lazy val rootJVM212 = project
+  .in(file("target/rootJVM212"))
+  .settings(
+    crossScalaVersions := Nil,
+    publish / skip     := true,
+    ideSkipProject     := true
+  )
+  .settings(crossScalaVersions := Nil)
```
Comment (ghostdogpr) [inline build.sbt]: This is duplicated (below as well)

**Review by ghostdogpr — COMMENTED** [review]
Ah that's nice. How about adding it to the instructions that show up when we start sbt? (for example the scala 3 case)

**Comment by kyri-petrou** [comment]
> Ah that's nice. How about adding it to the instructions that show up when we start sbt? (for example the scala 3 case)

Done - Also made some small changes to it that I think make sense - let me know what you think

File: .circleci/config.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
       - checkout
       - restore_cache:
           key: sbtcache
-      - run: sbt ++2.12.18 core/test http4s/test akkaHttp/test pekkoHttp/test zioHttp/test quickAdapter/test catsInterop/test tools/test codegenSbt/test clientJVM/test monixInterop/compile tapirInterop/test federation/test reporting/test tracing/test
+      - run: sbt ++2.12.18 rootJVM212/compile rootJVM212/test
```
Comment (ghostdogpr) [inline .circleci/config.yml]: Minor: I don't think you need to specify both, `test` will trigger `compile`

**Review by ghostdogpr — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (ghostdogpr) [reply .circleci/config.yml]: I changed it

**Pull request merged**